tests/wpt/mozilla/tests/mozilla/bluetooth/connect/connection-succeeds.html

```diff
@@ -0,0 +1,16 @@
+<!doctype html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/bluetooth/bluetooth-helpers.js"></script>
+<script>
+'use strict';
+promise_test(() => {
+    window.testRunner.setBluetoothMockDataSet(adapter_type.heart_rate);
+    return window.navigator.bluetooth.requestDevice({
+        filters: [{services: [heart_rate.name]}]
+    })
+    .then(device => device.gatt.connect())
+    .then(gattServer => assert_true(gattServer.connected))
+    .catch(error => assert_unreached('Promise should have resolved. ' + error));
```

Comment:
I believe `promise_test` automatically fails tests for unexpected rejections, so this can be removed from all the tests.